RavenM.Updater/Plugin.cs

```diff
@@ -33,55 +52,70 @@ private void Awake()
+            {
+                var response = MakeRequest("https://api.github.com/repos/iliadsh/RavenM/releases/latest");
+                var json = JSON.Parse(new StreamReader(response).ReadToEnd());
+                var asset = json["assets"][0];
+                time = asset["updated_at"];
+                download_url = asset["browser_download_url"];
+            }
+            else if (update_channel.Value == "dev")
+            {
+                var response = MakeRequest("https://api.github.com/repos/iliadsh/RavenM/actions/artifacts?name=RavenM&per_page=1");
+                var json = JSON.Parse(new StreamReader(response).ReadToEnd());
+                var artifact = json["artifacts"][0];
+                time = artifact["updated_at"];
+                download_url = artifact["archive_download_url"];
+            }
+            else
+            {
+                Logger.LogError($"Update channel {update_channel.Value} not found");
+                return;
+            }
 
-            foreach (var kv in json.AsArray)
+            DateTime upload_time;
+            if (DateTime.TryParse(time, out upload_time))
             {
-                var release = kv.Value;
-                var tag = release["tag_name"];
-                if (Version.TryParse(tag, out Version release_version))
+                if (upload_time > mod_creation_time)
                 {
-                    if (release_version > mod_version)
-                    {
-                        Logger.LogWarning($"Newer version found: {tag}. Downloading...");
+                    Logger.LogWarning($"Newer version found: {time}. Downloading...");
 
-                        var download_url = release["assets"][0]["browser_download_url"];
-                        using var client = new WebClient();
-                        using var zip = new MemoryStream(client.DownloadData(download_url));
+                    var zip = MakeRequest(download_url);
 
-                        Logger.LogWarning("Downloaded new version. Updating...");
+                    Logger.LogWarning("Downloaded new version. Updating...");
 
-                        using var archive = new ZipArchive(zip, ZipArchiveMode.Read);
-                        foreach (var s in archive.Entries)
+                    var archive = new ZipArchive(zip, ZipArchiveMode.Read);
+                    foreach (var s in archive.Entries)
+                    {
+                        // We can't update the updater, since it's already locked by BepInEx.
+                        if (s.Name != "RavenM.Updater.dll")
                         {
-                            // We can't update the updater, since it's already locked by BepInEx.
-                            if (s.Name != "RavenM.Updater.dll")
+                            try
+                            {
+                                s.ExtractToFile(PLUGIN_DIR + s.Name, true);
+                                File.SetCreationTime(PLUGIN_DIR + s.Name, DateTime.Now);
```

Comment:
Won't this make switching between release and dev awkward? After this is set, none of the currently uploaded builds on either channel would be able to beat it and get installed.

Also related: how would someone be able to switch off from dev if the builds are (almost) always newer than release?

Reply:
I'm thinking how to fix it